plugins/modules/idrac_diagnostics.py

```diff
@@ -714,6 +716,17 @@ def perform_check_mode(self):
             if 'SYS099' in message_id:
                 self.module.exit_json(msg=NO_FILE, skipped=True)
 
+    def expand_ipv6(self, ip):
```

Comment:
you could use ipaddress.IPv6Address(hostname).exploded

Reply:
I could have but I thought not of importing any libraries. That's why created one function for it.